pkg/locker/locker.go

```diff
@@ -90,6 +93,13 @@ func (l *Locker) Lock(name string) {
 	// once locked then we can decrement the number of waiters for this lock
 	nameLock.Lock()
 	nameLock.dec()
+
+	if nameLock.err != nil {
```

Comment:
The timeline is
```plain
goroutine1      goroutine2
   Lock
                [line85]nameLock.inc()    <- Now, the error has not been set.
                [line94]Lock, start to wait
                [line94]... wait
                [line94]... wait
   CancelWithError
   Unlock
                [line94] Finally get the lock.      <- Now, should check the error.
                Find the error. Give up the Lock. Unlock immediately.
   
```
Before line 94 goroutine2 really getting the lock, during the whole time, other goroutines all have chance to set an error into `l.lockers[name].err`. At line 85, no one has set the error yet.